src/lib/structures/ty.mli

```diff
@@ -58,9 +58,9 @@ type t =
   (** Enumeration, with its name, and the list of its constructors. *)
 
   | Tadt of Hstring.t * t list
-  (** Algebraic types applied to arguments. [Tadt (s, args)] is
-      the application of the datatype constructor [s] to
-      arguments [args]. *)
+  (** Application of algebraic data types. [Tadt (a, params)] denotes
+      the application of the polymorphim datatype [a] to the types parameters
```

Comment:
```suggestion
      the application of the polymorphic datatype [a] to the types parameters
```

The existing documentation is not technically wrong if you read "datatype constructor" as "(data) *type constructor*" but it is confusing and the new version is more clear (even better would be to include `Tadt ("List", ["Int"])` to represent `(List Int)` as an example).

Reply:
The documentation was really confusing for me and I wrote it! I added an example. Thanks :)